gradle-conjure/src/main/java/com/palantir/gradle/conjure/GradleExecUtils.java

```diff
@@ -47,4 +95,52 @@ static void exec(Project project, String failedTo, List<String> unloggedArgs, Li
                     failedTo, combinedArgs, execResult.getExitValue(), output.toString()));
         }
     }
+
+    private static void runJavaCodeInProcess(String failedTo, List<String> combinedArgs, Method mainMethod) {
+        try {
+            String[] args = combinedArgs.toArray(new String[] {});
+            mainMethod.invoke(null, new Object[] {args});
+        } catch (Throwable t) {
+            Throwable rootCause = Throwables.getRootCause(t);
+            if (rootCause instanceof GradleExecStubs.ExitInvoked) {
+                int exitStatus = ((GradleExecStubs.ExitInvoked) rootCause).getExitStatus();
+                if (exitStatus == 0) {
+                    // Exit status zero, we're good to go!
+                    return;
+                }
+                // the error message from a generator attempting to call exit 1 looks pretty gross
+                throw new RuntimeException(String.format(
+                        "Failed to %s. The command '%s' failed with exit code 1. Output above.",
+                        failedTo, combinedArgs));
+            }
+            throw new RuntimeException(
+                    String.format("Failed to %s. The command '%s' failed.", failedTo, combinedArgs), t);
+        }
+    }
+
+    private static Optional<Method> getMainMethod(URLClassLoader classLoader, String mainClassName) {
+        try {
+            ClassFileLocator locator = new ClassFileLocator.ForUrl(classLoader.getURLs());
+            TypePool typePool = TypePool.ClassLoading.of(classLoader);
+            Class<?> mainClass = new ByteBuddy(ClassFileVersion.ofThisVm(ClassFileVersion.JAVA_V8))
+                    .redefine(typePool.describe(mainClassName).resolve(), locator)
```

Comment:
I had not idea I could solve all my problems by just calling `redefine` this is incredible I love it